Owncloud iOs Client/Login/Login/UniversalLoginViewController.swift

```diff
@@ -0,0 +1,228 @@
+ You should have received a copy of this license
+ along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ */
+
+import Foundation
+
+struct K {
+    struct segueId {
+        static let segueToWebLoginView = "segueToWebLoginView"
+    }
+    
+    struct unwindId {
+        static let unwindToMainLoginView = "unwindToMainLoginView"
+    }
+    
+    struct vcId {
+        static let vcIdWebViewLogin = "WebViewLoginViewController"
+    }
+    
+}
+
+@objc class UniversalLoginViewController: UIViewController, UITextFieldDelegate, CheckAccessToServerDelegate {
+ 
+// MARK: IBOutlets
+    
+    @IBOutlet var imageViewLogo: UIImageView!
+    @IBOutlet var textFieldURL: UITextField!
+    @IBOutlet var buttonConnect: UIButton!
+    @IBOutlet var buttonHelpLink: UIButton!
+    @IBOutlet var buttonReconnection: UIButton!
+    
+    var urlNormalized: String!
+    var allAvailableAuthMethods = [AuthenticationMethod]()
+    var authMethodToLogin: AuthenticationMethod!
+    var authCodeReceived = ""
+
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        textFieldURL.delegate = self;
+        self.buttonConnect.isEnabled = false
+        // Do any additional setup after loading the view.
+        
+        //set branding style
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+
+    
+// MARK: checkUrl
+    
+    func checkCurrentUrl() {
+        
+        textFieldURL.resignFirstResponder()
+
+       //let stringURL = textFieldURL.text
+    
+        self.urlNormalized = textFieldURL.text //TODO: normalize url
+        //parse url, appsfiles, user:pass, prefix
+        
+        
+        //check status and get authmethods available
+        
+        let checkAccessToServer : CheckAccessToServer = CheckAccessToServer.sharedManager() as! CheckAccessToServer
+        checkAccessToServer.delegate = self
+        checkAccessToServer.isConnectionToTheServer(byUrl: self.urlNormalized)
+        
+    }
+    
+    
+// MARK: start log in auth
+    
+    func startAuthenticationWith(authMethod: AuthenticationMethod) {
+        
+        switch authMethod {
+        case .SAML_WEB_SSO:
+
+            LoginViewController().showSSOLoginScreen(withUrl: self.urlNormalized, inVC: self)
+            
+            break
+        case .BEARER_TOKEN:
+            
+            performSegue(withIdentifier: K.segueId.segueToWebLoginView, sender: self)
+            
+            break
+        case .BASIC_HTTP_AUTH:
+            //TODO
+            break
+        default:
+            //TODO: show footer Error
+
+            break
+        }
+    }
+    
+    func openWebViewLogin() {
+        
+    }
+    
+    
+// MARK:  CheckAccessToServer delegate
+    
+    func connection(toTheServer isConnection: Bool) {
```

Comment:
👍 but very careful with the refactor.. it it's used in a lot of places in the app